File: examples/android/ExampleApp.java
package com.londonbridge.ai.example;

import android.app.Application;
import android.util.Log;

import com.londonbridge.ai.memory.MemoryEntry;
import com.londonbridge.ai.memory.SecureMemoryManager;

import java.util.HashMap;
import java.util.Map;

/**
 * Example Android application demonstrating London Bridge AI integration
 */
public class ExampleApp extends Application {
    private static final String TAG = "ExampleApp";
    private SecureMemoryManager memoryManager;
    
    @Override
    public void onCreate() {
        super.onCreate();
        
        try {
            // Initialize memory manager
            memoryManager = new SecureMemoryManager(this);
            Log.i(TAG, "Memory manager initialized");
            
            // Example: Store a conversation memory
            storeConversationExample();
            
            // Example: Retrieve and use a memory
            retrieveMemoryExample();
            
            // Example: Check memory statistics
            checkStatsExample();
            
        } catch (Exception e) {
            Log.e(TAG, "Failed to initialize memory manager", e);
        }
    }
    
    private void storeConversationExample() {
        Map<String, String> metadata = new HashMap<>();
        metadata.put("type", "conversation");
        metadata.put("model", "gpt-4");
        metadata.put("user_id", "user_123");
        metadata.put("topic", "weather");
        
        String interactionId = "conversation_" + System.currentTimeMillis();
        String content = "User: What's the weather like?\n" +
                        "Assistant: I checked the forecast and it will be sunny with " +
                        "a high of 75°F today. Perfect weather for outdoor activities!";
        
        boolean success = memoryManager.storeMemory(interactionId, content, metadata);
        
        if (success) {
            Log.i(TAG, "Stored conversation memory: " + interactionId);
        } else {
            Log.e(TAG, "Failed to store conversation memory");
        }
    }
    
    private void retrieveMemoryExample() {
        // In a real app, you'd retrieve a specific interaction ID
        // For demo purposes, we'll just show the pattern
        
        String interactionId = "conversation_12345";
        MemoryEntry entry = memoryManager.retrieveMemory(interactionId);
        
        if (entry != null) {
            // Verify integrity
            if (entry.verifyIntegrity()) {
                Log.i(TAG, "Retrieved memory: " + entry.getId());
                Log.i(TAG, "Content: " + entry.getContent());
                Log.i(TAG, "Timestamp: " + entry.getTimestamp());
                
                // Use the memory to provide context to AI
                // String context = entry.getContent();
                // aiModel.setContext(context);
                
            } else {
                Log.w(TAG, "Memory integrity check failed!");
            }
        } else {
            Log.i(TAG, "Memory not found: " + interactionId);
        }
    }
    
    private void checkStatsExample() {
        SecureMemoryManager.MemoryStats stats = memoryManager.getMemoryStats();
        
        Log.i(TAG, "Memory Statistics:");
        Log.i(TAG, "  Total memories: " + stats.count);
        Log.i(TAG, "  Total size: " + (stats.totalSizeBytes / 1024) + " KB");
        
        // Alert if getting close to limit
        long maxSizeBytes = 100 * 1024 * 1024; // 100 MB
        double usagePercent = (stats.totalSizeBytes * 100.0) / maxSizeBytes;
        
        if (usagePercent > 80) {
            Log.w(TAG, "Memory usage at " + String.format("%.1f%%", usagePercent));
        }
    }
    
    /**
     * Example: Store AI learning moment
     */
    public void storeAILearning(String topic, String insight) {
        Map<String, String> metadata = new HashMap<>();
        metadata.put("type", "learning");
        metadata.put("topic", topic);
        metadata.put("timestamp", String.valueOf(System.currentTimeMillis()));
        
        String interactionId = "learning_" + topic + "_" + System.currentTimeMillis();
        memoryManager.storeMemory(interactionId, insight, metadata);
    }
    
    /**
     * Example: Retrieve recent memories for context
     */
    public void getRecentContext(long timeRangeMs) {
        long endTime = System.currentTimeMillis();
        long startTime = endTime - timeRangeMs;
        
        var memories = memoryManager.retrieveMemoriesInRange(startTime, endTime);
        
        Log.i(TAG, "Found " + memories.size() + " recent memories");
        
        // Build context from recent memories
        StringBuilder context = new StringBuilder();
        for (MemoryEntry entry : memories) {
            context.append(entry.getContent()).append("\n\n");
        }
        
        // Provide to AI model
        // aiModel.setContext(context.toString());
    }
}

File: examples/ios/ExampleApp.swift
//
//  ExampleApp.swift
//  LondonBridgeAI Example
//
//  Example iOS application demonstrating London Bridge AI integration
//

import UIKit

@available(iOS 13.0, *)
@main
class ExampleApp: UIApplication {
    
    private var memoryManager: SecureMemoryManager?
    
    override init() {
        super.init()
        
        do {
            // Initialize memory manager
            memoryManager = try SecureMemoryManager()
            print("[ExampleApp] Memory manager initialized")
            
            // Example: Store a conversation memory
            storeConversationExample()
            
            // Example: Retrieve and use a memory
            retrieveMemoryExample()
            
            // Example: Check memory statistics
            checkStatsExample()
            
        } catch {
            print("[ExampleApp] Failed to initialize memory manager: \(error)")
        }
    }
    
    private func storeConversationExample() {
        guard let manager = memoryManager else { return }
        
        let metadata = [
            "type": "conversation",
            "model": "gpt-4",
            "user_id": "user_123",
            "topic": "weather"
        ]
        
        let interactionId = "conversation_\(Int(Date().timeIntervalSince1970))"
        let content = """
        User: What's the weather like?
        Assistant: I checked the forecast and it will be sunny with a high of 75°F today. Perfect weather for outdoor activities!
        """
        
        let success = manager.storeMemory(
            interactionId: interactionId,
            content: content,
            metadata: metadata
        )
        
        if success {
            print("[ExampleApp] Stored conversation memory: \(interactionId)")
        } else {
            print("[ExampleApp] Failed to store conversation memory")
        }
    }
    
    private func retrieveMemoryExample() {
        guard let manager = memoryManager else { return }
        
        // In a real app, you'd retrieve a specific interaction ID
        // For demo purposes, we'll just show the pattern
        
        let interactionId = "conversation_12345"
        
        if let entry = manager.retrieveMemory(interactionId: interactionId) {
            // Verify integrity
            if entry.verifyIntegrity() {
                print("[ExampleApp] Retrieved memory: \(entry.id)")
                print("[ExampleApp] Content: \(entry.content)")
                print("[ExampleApp] Timestamp: \(entry.timestamp)")
                
                // Use the memory to provide context to AI
                // let context = entry.content
                // aiModel.setContext(context)
                
            } else {
                print("[ExampleApp] Memory integrity check failed!")
            }
        } else {
            print("[ExampleApp] Memory not found: \(interactionId)")
        }
    }
    
    private func checkStatsExample() {
        guard let manager = memoryManager else { return }
        
        let stats = manager.getMemoryStats()
        
        print("[ExampleApp] Memory Statistics:")
        print("[ExampleApp]   Total memories: \(stats.count)")
        print("[ExampleApp]   Total size: \(stats.totalSizeBytes / 1024) KB")
        
        // Alert if getting close to limit
        let maxSizeBytes: Int64 = 100 * 1024 * 1024 // 100 MB
        let usagePercent = Double(stats.totalSizeBytes) * 100.0 / Double(maxSizeBytes)
        
        if usagePercent > 80 {
            print("[ExampleApp] Memory usage at \(String(format: "%.1f%%", usagePercent))")
        }
    }
    
    /// Example: Store AI learning moment
    func storeAILearning(topic: String, insight: String) {
        guard let manager = memoryManager else { return }
        
        let metadata = [
            "type": "learning",
            "topic": topic,
            "timestamp": String(Int(Date().timeIntervalSince1970))
        ]
        
        let interactionId = "learning_\(topic)_\(Int(Date().timeIntervalSince1970))"
        _ = manager.storeMemory(interactionId: interactionId, content: insight, metadata: metadata)
    }
    
    /// Example: Retrieve recent memories for context
    func getRecentContext(timeRangeSeconds: TimeInterval) {
        guard let manager = memoryManager else { return }
        
        let endTime = Date().timeIntervalSince1970
        let startTime = endTime - timeRangeSeconds
        
        let memories = manager.retrieveMemories(inRange: startTime, endTime: endTime)
        
        print("[ExampleApp] Found \(memories.count) recent memories")
        
        // Build context from recent memories
        var context = ""
        for entry in memories {
            context += entry.content + "\n\n"
        }
        
        // Provide to AI model
        // aiModel.setContext(context)
    }
}

@available(iOS 13.0, *)
class AppDelegate: UIResponder, UIApplicationDelegate {
    
    func application(_ application: UIApplication,
                    didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
        
        // Start persistence manager
        let persistenceManager = MemoryPersistenceManager.shared
        persistenceManager.start()
        
        return true
    }
    
    func applicationWillTerminate(_ application: UIApplication) {
        // Perform emergency backup
        let persistenceManager = MemoryPersistenceManager.shared
        persistenceManager.performEmergencyBackup()
    }
}
